### PR TITLE
Remove commitment shifting, but only in proof-systems (Part 1)

### PR DESCRIPTION
**Pull request by volhovm**
This addresses https://github.com/MinaProtocol/mina/issues/14628

This PR uses the version of proof-systems without shifted commitments (https://github.com/o1-labs/proof-systems/pull/1377), however it does not remove them from `mina` (step 1), but merely asserts that this functionality is obsolete. 

Step 2, next PR, is https://github.com/MinaProtocol/mina/pull/14656 -- removing shifts from `mina` itself.

### CODE REVIEW

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs [hunk 1]
```diff
@@ -975,8 +969,7 @@ pub fn caml_pasta_fp_plonk_proof_dummy() -> CamlProofWithPublic<CamlGVesta, Caml
     fn comm() -> PolyComm<Vesta> {
         let g = Vesta::prime_subgroup_generator();
         PolyComm {
-            shifted: Some(g),
```
Comment (fabrizio-m) [inline src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs:978]: I don't know where this is used, but doesn't the behavior change if before it was shift and now isn't?

**Review by fabrizio-m — APPROVED** [review]
Just let a comment with regards to a change, not saying it is an issue, but additional clarification would be useful.

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs (hunk 1 above)
Comment (volhovm) [reply src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs:978]: I think it shouldn't -- wherever it's used, removing `shifted` does not seem to break serialization or tests. Will investigate before merging.

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

File: src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs (hunk 1 above)
Comment (volhovm) [reply src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs:978]: @fabrizio-m I left a comment in the main PR thread. TLDR I also don't know but it seems to be OK.

**Comment by mitschabaude** [comment]
> Given that there's quite some external evidence that `dummy` is not terribly important (maybe used by some js tests?)

JS only uses `Pickles.Proof.dummy`, which doesn't seem to use the dummy proof from Kimchi

**Pull request merged**